### PR TITLE
Add ES6 information to JavaScript file

### PR DESCRIPTION
**Pull request by davidbailey00**
- [x] PR touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at _intermediate to experienced programmers_ (this is a poor format for explaining fundamental programming concepts)
- [x] YAML Frontmatter formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Seriously, look at it now. Watch for quotes and double-check field names.

### CODE REVIEW

File: javascript.html.markdown [hunk 1]
```diff
@@ -201,7 +220,34 @@ myObj.myThirdKey = true;
...
+// You can also write the above as follows:
+var myNumber = 42;
+export myNumber;
+
+// To import variables from another script, use `import`.
+import * as numbers from "numbers"; // file imported is numbers.js
+numbers.myNumber === 42; // true
+
+// You can also specific variables/functions.
+import myNumber from "numbers";
```
Comment (geoffliu) [inline javascript.html.markdown]: `import { myNumber } ...`

File: javascript.html.markdown [hunk 2]
```diff
@@ -201,7 +220,34 @@ myObj.myThirdKey = true;
...
+
+// You can also write the above as follows:
+var myNumber = 42;
+export myNumber;
+
+// To import variables from another script, use `import`.
+import * as numbers from "numbers"; // file imported is numbers.js
+numbers.myNumber === 42; // true
+
+// You can also specific variables/functions.
```
Comment (geoffliu) [inline javascript.html.markdown]: you can also import specific...

File: javascript.html.markdown [hunk 3]
```diff
@@ -201,7 +220,34 @@ myObj.myThirdKey = true;
...
+import myNumber from "numbers";
+myNumber === 42; // true
+
+// When importing multiple specific variables, surround them with curly braces.
```
Comment (geoffliu) [inline javascript.html.markdown]: You always need curly braces for the kind of import introduced here. You're not mentioning default exports, which is what gets imported without the curly braces.

File: javascript.html.markdown [hunk 4]
```diff
@@ -201,7 +220,34 @@ myObj.myThirdKey = true;
...
+import {myNumber, otherNumber} from "numbers";
+myNumber + otherNumber === 105; // true
+
+// Remember that modules using the syntax shown above are only supported in ES6
```
Comment (geoffliu) [inline javascript.html.markdown]: Currently there aren't many environments out there that support import/export syntax directly. Most environments will require a Babel transpile. Please clarify.

File: javascript.html.markdown [hunk 5]
```diff
@@ -284,14 +330,29 @@ switch (grade) {
...
-// 4. Functions, Scope and Closures
+// 5. Functions, Scope and Closures
 
 // JavaScript functions are declared with the `function` keyword.
 function myFunction(thing){
     return thing.toUpperCase();
 }
 myFunction("foo"); // = "FOO"
 
+// You can specify a default argument if one isn't given in ES6 environments.
```
Comment (geoffliu) [inline javascript.html.markdown]: Comma after "given"

File: javascript.html.markdown [hunk 6]
```diff
@@ -284,14 +330,29 @@ switch (grade) {
...
+function sayHello(name="world") {
+    alert(`Hello, ${name}!`);
+}
+sayHello(); // Hello, world!
+sayHello("Alice"); // Hello, Alice!
+
+// ES6 environments also let you use an array items as arguments to a function.
```
Comment (geoffliu) [inline javascript.html.markdown]: an array's items

File: javascript.html.markdown [hunk 7]
```diff
@@ -323,6 +384,19 @@ setTimeout(function(){
     // this code will be called in 5 seconds' time
 }, 5000);
 
+// You can also write anonymous functions using arrow syntax in ES6
+// environments. The following is the same as the above.
+setTimeout(() => {
+    // this code will be called in 5 seconds' time
+}, 5000);
+
+// When there is exactly one argument, brackets are not needed.
```
Comment (geoffliu) [inline javascript.html.markdown]: s/brackets/parentheses/

File: javascript.html.markdown [hunk 8]
```diff
@@ -323,6 +384,19 @@ setTimeout(function(){
     // this code will be called in 5 seconds' time
 }, 5000);
 
+// You can also write anonymous functions using arrow syntax in ES6
+// environments. The following is the same as the above.
```
Comment (geoffliu) [inline javascript.html.markdown]: Well, no. They're not the same. Arrow functions do not introduce a new context, while `function`s do.

File: javascript.html.markdown [hunk 9]
```diff
@@ -438,6 +519,20 @@ var MyConstructor = function(){
 myNewObj = new MyConstructor(); // = {myNumber: 5}
 myNewObj.myNumber; // = 5
 
+// In ES6 environments, there is a more simple way to create classes.
```
Comment (geoffliu) [inline javascript.html.markdown]: simpler

File: javascript.html.markdown [hunk 10]
```diff
@@ -438,6 +519,20 @@ var MyConstructor = function(){
...
+
+class MyConstructor {
```
Comment (geoffliu) [inline javascript.html.markdown]: Let's call this `MyClass` instead

**Comment by geoffliu** [comment]
@vendethiel I forget, didn't we decide to have a separate page for ES6, like we did with the Python 2/3 split?

**Comment by geoffliu** [comment]
Ah, we're still unsure... https://github.com/adambard/learnxinyminutes-docs/pull/1905 😭

**Comment by vendethiel** [comment]
Hey!

I think it's about time we start adding ES6 info.
There's another PR for it: #1905 .
I'm not sure whichever should be picked, but if we could get them "merged" I think it'd be pretty great.

**Comment by divayprakash** [comment]
Closed in favor of #2858.

**Pull request closed without merging**